### PR TITLE
leo_gateway: Added Parital Read Write Test for NFS

### PR DESCRIPTION
**Pull request by windkit**
## Related Issue
https://github.com/leo-project/leofs/issues/601

### CODE REVIEW

File: apps/leo_gateway/test/leo_nfs_integration_tests.sh [hunk 1]
```diff
@@ -173,6 +195,8 @@ function check_dummy() {
     ls_validate_num_of_childs $MOUNT_DIR 504 &&
     eval $CMD_DD &&
     stat_validate_size $DST_50M_FILE4 128 &&
+    partial_write_test $TMP_50M_FILE $DST_50M_FILE3 524288 &&
+    partial_read_test $TMP_50M_FILE $DST_50M_FILE3 1572864 1048576 &&
```
Comment (mocchira) [inline apps/leo_gateway/test/leo_nfs_integration_tests.sh]: As this test is supposed to run on not only linux but other xunix we are supporting (freebsd/smartos,
We need to do either
1. Make drop_cache work on others (seems to be difficult but not impossible
2. Skip partial_(read|write)_test on others

For now, let's go with 2

File: apps/leo_gateway/test/leo_nfs_integration_tests.sh [hunk 2]
```diff
@@ -141,6 +138,45 @@ function check_dummy() {
...
+        TMP_FILE=$1
+        TAR_FILE=$2
+        OFFSET=$3
+        dd if=$TMP_1M_FILE of=$TMP_FILE bs=$OFFSET conv=notrunc seek=1
+        dd if=$TMP_1M_FILE of=$TAR_FILE bs=$OFFSET conv=notrunc seek=1
+        drop_cache
+        diff $TMP_FILE $TAR_FILE
+        ;;
+    *)
+        return 1
```
Comment (mocchira) [inline apps/leo_gateway/test/leo_nfs_integration_tests.sh]: return 0 to skip

File: apps/leo_gateway/test/leo_nfs_integration_tests.sh [hunk 3]
```diff
@@ -141,6 +138,45 @@ function check_dummy() {
...
+        TAR_FILE=$2
+        OFFSET=$3
+        SIZE=$4
+        drop_cache
+        dd if=$TMP_FILE of=$TMP_DIR/part_read_ori skip=$OFFSET bs=$SIZE count=1 iflag=skip_bytes
+        dd if=$TAR_FILE of=$TMP_DIR/part_read_tar skip=$OFFSET bs=$SIZE count=1 iflag=skip_bytes
+        diff $TMP_DIR/part_read_ori $TMP_DIR/part_read_tar
+        ;;
+    *)
+        return 1
```
Comment (mocchira) [inline apps/leo_gateway/test/leo_nfs_integration_tests.sh]: return 0 to skip

**Comment by windkit** [comment]
@mocchira Updated.

**Pull request merged**